cdc/processor/pipeline/sorter.go

```diff
@@ -160,6 +152,10 @@ func (n *sorterNode) StartActorNode(ctx pipeline.NodeContext, isTableActorMode b
 		for {
 			select {
 			case <-stdCtx.Done():
+				// Since the flowController is implemented by `Cond`, it is not cancelable
+				// by a context. We need to listen on cancellation and aborts the flowController
+				// manually.
+				n.flowController.Abort()
```

Comment:
The current implementation of `flowController` cannot be canceled by contexts, so the `Abort()` may never be called, causing a deadlock.

Reply:
fixed